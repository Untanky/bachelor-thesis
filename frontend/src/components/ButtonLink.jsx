import React from 'react';
import { Link } from 'react-router-dom';

import './ButtonLink.scss';

const Button = ({ to, text }) => {
  return (
    <Link className="button-link" to={to}>
      {text}
    </Link>
  )
};

export default Button;
