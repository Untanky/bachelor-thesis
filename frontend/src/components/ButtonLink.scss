@import '../theme';

.button-link {
  background-color: $htw-green;
  color: white;
  text-decoration: none;
  outline: none;
  padding: 0.5em 1em;
  border-radius: 2em;
  box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.2);
  transition: all 0.25s ease-in-out;

  &:hover {
    cursor: pointer;
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.15);
    transition: all 0.25s ease-in-out;
  }
}