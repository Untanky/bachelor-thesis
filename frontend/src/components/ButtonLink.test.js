import React from 'react';
import ButtonLink from './ButtonLink';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <ButtonLink to="/home" text="Homepage" />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
