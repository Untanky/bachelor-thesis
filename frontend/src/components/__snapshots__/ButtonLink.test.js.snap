// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="button-link"
  href="/home"
  onClick={[Function]}
>
  Homepage
</a>
`;
